lib/puppet/resource/catalog.rb

```diff
@@ -86,13 +108,12 @@ def container_of(resource)
     adjacent(resource, :direction => :in)[0]
   end
 
-  def add_one_resource(resource)
-    title_key = title_key_for_ref(resource.ref)
+  def add_one_resource(resource, title_key, idx)
```

Comment:
I ended up needing the title_key in my functions that call this, so I wanted to just pass it through instead of fetching it multiple times.